deepchem/models/torch_models/grover_layers.py

```
@@ -0,0 +1,688 @@
+    """
+
+    # FIXME This layer is similar to DMPNNEncoderLayer and they
+    # must be unified.
+    def __init__(self,
+                 atom_messages: bool,
+                 init_message_dim: int,
+                 hidden_size: int,
+                 depth: int,
+                 undirected: bool,
+                 attach_feats: bool,
+                 attached_feat_fdim: int = 0,
+                 bias: bool = True,
+                 dropout: float = 0.2,
+                 activation: str = 'relu',
+                 input_layer: str = 'fc',
+                 dynamic_depth: str = 'none'):
+        super(GroverMPNEncoder, self).__init__()
+        if input_layer == 'none':
+            assert init_message_dim == hidden_size
+        self.init_message_dim = init_message_dim
+        self.depth = depth
+        self.input_layer = input_layer
+        self.layers_per_message = 1
+        self.undirected = undirected
+        self.atom_messages = atom_messages
+        self.attached_feat = attach_feats
+
+        assert dynamic_depth in [
+            'none', 'truncnorm', 'uniform'
+        ], 'If dynamic depth, it should be truncnorm or uniform'
+        self.dynamic_depth = dynamic_depth
+
+        self.dropout_layer = nn.Dropout(p=dropout)
+        if activation == 'relu':
+            self.act_func = nn.ReLU()
+        else:
+            raise ValueError('Only ReLU activation function is supported')
+
+        if self.input_layer == "fc":
+            input_dim = self.init_message_dim
+            self.W_i = nn.Linear(input_dim, hidden_size, bias=bias)
+
+        # Bug here, if input_layer is none, then w_h_input_size should have init_messages_dim
+        # instead of hidden_size for first iteration of message passing alone (in subsequent
+        # iterations, message size will be n_bonds x hidden_size but during first iteration,
+        # message size will be init_messages_dim
```

Maybe raise an error in this case?